extensions/amp-tiktok/0.1/amp-tiktok.js

```diff
@@ -0,0 +1,265 @@
+              fill="#FF004F"
+            />
+            <path
+              d="M28.8505 14.1641C31.704 16.1997 35.1983 17.3975 38.9724 17.3975V11.6901C36.8657 11.2421 35.0007 10.143 33.5984 8.61433C31.1982 7.11909 29.4689 4.65232 28.9618 1.76514H23.6706V30.7258C23.6586 34.102 20.914 36.8357 17.5294 36.8357C15.5349 36.8357 13.763 35.8866 12.6408 34.4164C10.6369 33.4067 9.26266 31.3338 9.26266 28.94C9.26266 25.5544 12.0118 22.8095 15.4037 22.8095C16.0536 22.8095 16.68 22.9105 17.2675 23.0966V17.3268C9.9835 17.477 4.12537 23.4186 4.12537 30.7259C4.12537 34.3737 5.58418 37.6805 7.95184 40.0967C10.0878 41.5285 12.6591 42.3636 15.4256 42.3636C22.84 42.3636 28.8507 36.3631 28.8507 28.9611V14.1641H28.8505Z"
+              fill="black"
+            />
+            <path
+              d="M38.9726 11.69V10.1468C37.0728 10.1497 35.2104 9.61856 33.5986 8.61416C35.0254 10.1736 36.9042 11.2489 38.9726 11.69ZM28.962 1.76512C28.9136 1.48918 28.8765 1.21143 28.8507 0.932611V0H21.545V28.9609C21.5333 32.3367 18.7888 35.0704 15.4039 35.0704C14.4101 35.0704 13.4718 34.8349 12.6409 34.4165C13.7631 35.8866 15.535 36.8356 17.5296 36.8356C20.9139 36.8356 23.659 34.1021 23.6708 30.7259V1.76512H28.962ZM17.268 17.3268V15.6839C16.6575 15.6006 16.0421 15.5588 15.4259 15.5591C8.01082 15.5589 2 21.5594 2 28.9609C2 33.6013 4.36236 37.6908 7.95215 40.0964C5.58448 37.6803 4.12567 34.3733 4.12567 30.7257C4.12567 23.4186 9.98365 17.477 17.268 17.3268Z"
+              fill="#00F2EA"
+            />
+          </g>
+          <defs>
+            <clipPath id="clip0">
+              <rect width="44" height="44" fill="white" />
+            </clipPath>
+          </defs>
+        </svg>
+      `,
+    };
+  }
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {!Element} */
+    this.iframe_ = null;
+
+    /** @private {string} */
+    this.videoId_ = null;
+
+    /** @private {?Function}*/
+    this.unlistenMessage_ = null;
+
+    /** @private {?Promise} */
+    this.resolveReceivedFirstMessage_ = null;
+
+    /** @private {string} */
+    this.iframeNameString_ = this.getIframeNameString_();
+
+    /**
+     * @private {number}
+     * This value is calculated by multiplying our fixed width (325px)
+     * by the video aspect ratio (13:23 or 1.77) and adding 200px to account
+     * for the height of the caption.
+     */
+    this.fallbackHeight_ = 775;
```

Comment:
What if we encode this rule as a constant expression?

```suggestion
    /**
     * @private {number}
     * This value is calculated by multiplying our fixed width (325px)
     * by the video aspect ratio (13:23 or 1.77) and adding 200px to account
     * for the height of the caption.
     */
    this.fallbackHeight_ = WIDTH * ASPECT + CAPTION_HEIGHT;
```

Reply:
+1